lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Table.php

```diff
@@ -730,16 +730,17 @@ public function writeConstructor(WriterInterface $writer)
         foreach ($this->columns as $column) {
             if ($column->isPrimary()) {
                 $id = $column->getName();
+                $isUuid = $column->isUuid();
             }
         }
 
         $writer
-            ->writeIf($rasmeyUuidProvider, 'public function __construct(?string $uuid = null)')
-            ->writeIf(!$rasmeyUuidProvider, 'public function __construct()')
+            ->writeIf($rasmeyUuidProvider && $isUuid, 'public function __construct(?string $uuid = null)')
+            ->writeIf(!($rasmeyUuidProvider && $isUuid), 'public function __construct()')
             ->write('{')
             ->indent()
-            ->writeIf($rasmeyUuidProvider, '$uuid = $uuid ? $uuid : Uuid::uuid1()->toString();')
-            ->writeIf(($rasmeyUuidProvider && $id), '$this->'.$id.' = $uuid;')
+            ->writeIf($rasmeyUuidProvider && $isUuid, '$uuid = $uuid ? $uuid : Uuid::uuid1()->toString();')
```

Comment:
@peterchallett tested this on my current PR of generic bo https://github.com/OnCorps/bo-generic-base/pull/4 (which I will update after merge of this) and is working with sample data. 
It's also fair to say that the case for which we have auto generated incremental id and we don't want a set is solvable like follow: we annotate `@GeneratedValue` on the primary key if the id is not imported and is not uuid. Basically saying the fallback is always doctrine GeneratedValue. Then fixtures and tests can live with that as rather than attempting to set a known id they could simply rely on the sequence of appearance 
eg. writing this fixture data for a SolarSystem entity would be like this (omitting keys):
```
[
  {
    "entity": {
      "name": "This"
    },
    "purpose": "GET"
  },
  {
    "entity": {
      "name": "That"
    },
    "purpose": "DELETE"
  }
]
```
and is implicit that first element will have key=1 and second will have key=2 in test db (much more reliable than working with uuid)

Reply:
todo add `@GeneratedValue` annotation as fallback id strategy

Reply:
That's great but you don't need to add the @GeneratedValue as it is already supported:
![image](https://user-images.githubusercontent.com/3935346/61802136-e296c500-ae27-11e9-8f54-03e2313cda8d.png)

Reply:
And thanks for doing the PR.

Reply:
oh even better if that's already the case :)! (testing wise would be ok write tests against generated values that are not uuid as there are hooks speeding up and tearing down the test database at each scenario so auto increment will start from scratch at each run)